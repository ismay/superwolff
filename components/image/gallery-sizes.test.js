import calculateSizes from "./gallery-sizes";

const layouts = [
  { amount: 1, index: 0, number: 1 },
  { amount: 2, index: 1, number: 2 },
  { amount: 3, index: 2, number: 3 },
  { amount: 4, index: 3, number: 4 },
  { amount: 5, index: 4, number: 5 },
];

describe("calculateSizes", () => {
  it.each(layouts)(
    "should return the expected sizes for image $number of total $amount",
    ({ amount, index, number }) => {
      expect(index + 1).toBe(number);
      expect(calculateSizes({ amount, index })).toMatchSnapshot();
    }
  );
});
