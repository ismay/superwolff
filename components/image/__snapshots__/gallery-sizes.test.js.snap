// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateSizes should return the expected sizes for image 1 of total 1 1`] = `"(min-width: 50em) 50em, (min-width: 35em) 100vw, (min-width: 30em) 100vw, 100vw"`;

exports[`calculateSizes should return the expected sizes for image 2 of total 2 1`] = `"(min-width: 50em) 25em, (min-width: 35em) 50vw, (min-width: 30em) 50vw, 100vw"`;

exports[`calculateSizes should return the expected sizes for image 3 of total 3 1`] = `"(min-width: 50em) 17em, (min-width: 35em) 33vw, (min-width: 30em) 100vw, 100vw"`;

exports[`calculateSizes should return the expected sizes for image 4 of total 4 1`] = `"(min-width: 50em) 50em, (min-width: 35em) 100vw, (min-width: 30em) 50vw, 100vw"`;

exports[`calculateSizes should return the expected sizes for image 5 of total 5 1`] = `"(min-width: 50em) 25em, (min-width: 35em) 50vw, (min-width: 30em) 100vw, 100vw"`;
